src/mcpwm/operator_config.rs

```diff
@@ -0,0 +1,151 @@
+
+    /// Configuration for Comparator Y
+    pub(crate) comparator_y: CMPY,
+
+    /// Configuration for Generator A
+    pub(crate) generator_a: GENA,
+
+    /// Configuration for Generator B
+    pub(crate) generator_b: GENB,
+}
+
+impl OperatorConfig<NoCmpCfg, NoCmpCfg, NoGenCfg, NoGenCfg> {
+    pub fn default<PA: OutputPin, PB: OutputPin>(
+        pin_a: PA,
+        pin_b: PB,
+    ) -> OperatorConfig<
+        ComparatorConfig,
+        ComparatorConfig,
+        DefaultGeneratorConfigA<PA>,
+        DefaultGeneratorConfigB<PB>,
+    > {
+        OperatorConfig::empty()
+            .cmp_x(Default::default())
+            .cmp_y(Default::default())
+            .gen_a(GeneratorConfig::active_high(pin_a))
+            .gen_b(GeneratorConfig::active_high(pin_b))
+    }
+
+    pub fn empty() -> Self {
+        let mut flags: mcpwm_operator_config_t__bindgen_ty_1 = Default::default();
+
+        // TODO: What value makes most sense as default here?
+        flags.set_update_gen_action_on_tez(1);
+        flags.set_update_gen_action_on_tep(1);
+        flags.set_update_gen_action_on_sync(1);
+
+        flags.set_update_dead_time_on_tez(1);
+        flags.set_update_dead_time_on_tep(1);
+        flags.set_update_dead_time_on_sync(1);
+
+        OperatorConfig {
+            flags,
+            comparator_x: NoCmpCfg, // SOC_MCPWM_COMPARATORS_PER_OPERATOR is 2 for ESP32 and ESP32-S3
+            comparator_y: NoCmpCfg,
+
+            generator_a: NoGenCfg, // One generator per pin, with a maximum of two generators per Operator
+            generator_b: NoGenCfg,
+        }
+    }
+}
+
+impl<CMPY: OptionalCmpCfg> OperatorConfig<NoCmpCfg, CMPY, NoGenCfg, NoGenCfg> {
+    fn cmp_x(
+        self,
+        config: ComparatorConfig,
+    ) -> OperatorConfig<ComparatorConfig, CMPY, NoGenCfg, NoGenCfg> {
+        OperatorConfig {
+            flags: self.flags,
+            comparator_x: config,
+            comparator_y: self.comparator_y,
+
+            generator_a: self.generator_a,
+            generator_b: self.generator_b,
+        }
+    }
+}
+
+impl<CMPX: OptionalCmpCfg> OperatorConfig<CMPX, NoCmpCfg, NoGenCfg, NoGenCfg> {
+    fn cmp_y(
+        self,
+        config: ComparatorConfig,
+    ) -> OperatorConfig<CMPX, ComparatorConfig, NoGenCfg, NoGenCfg> {
+        OperatorConfig {
+            flags: self.flags,
+            comparator_x: self.comparator_x,
+            comparator_y: config,
+
+            generator_a: self.generator_a,
+            generator_b: self.generator_b,
+        }
+    }
+}
+
+impl<CMPX: OptionalCmpCfg, CMPY: OptionalCmpCfg, GENB: OptionalGenCfg>
+    OperatorConfig<CMPX, CMPY, NoGenCfg, GENB>
+{
+    #[allow(clippy::type_complexity)]
+    fn gen_a<P: OutputPin>(
+        self,
+        config: GeneratorConfig<GenA, CMPX::OnMatchCfg, CMPY::OnMatchCfg, P>,
+    ) -> OperatorConfig<
+        CMPX,
+        CMPY,
+        GeneratorConfig<GenA, CMPX::OnMatchCfg, CMPY::OnMatchCfg, P>,
+        GENB,
+    > {
```

Comment:
Is there any way to make clippy happy about such a type?

Reply:
I think silencing clippy here is okay :).